src/hypothesis/strategies.py

```diff
@@ -950,29 +950,71 @@ def fractions(min_value=None, max_value=None, max_denominator=None):
     max_denominator must be at least 1.
 
     """
+    if min_value is not None:
+        try:
+            min_value = Fraction(min_value)
+        except (TypeError, ValueError):
+            raise InvalidArgument('min_value=%r to Fraction' % (min_value,))
+    if max_value is not None:
+        try:
+            max_value = Fraction(max_value)
+        except (TypeError, ValueError):
+            raise InvalidArgument('max_value=%r to Fraction' % (max_value,))
+        if min_value == max_value:
+            return just(min_value)
+
     check_valid_bound(min_value, 'min_value')
     check_valid_bound(max_value, 'max_value')
     check_valid_interval(min_value, max_value, 'min_value', 'max_value')
 
     check_valid_integer(max_denominator)
     if max_denominator is not None and max_denominator < 1:
+        raise InvalidArgument('Denominator=%r must be >= 1' % max_denominator)
+
+    def clip_denominator(frac):
+        """Clip fraction to a max denominator, and bump back over bounds."""
+        if max_denominator is None:
+            return frac
+        val = frac.limit_denominator(max_denominator)
+        if min_value is not None and val < min_value:
```

Comment:
Actually I don't entirely understand why the `limit_denominator` does anything at all. Doesn't `dm_func` below ensure we're in the right denominator range?